### PR TITLE
Added a function UpdateMaintenanceCronJob

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:

- [x] Added the function `UpdateMaintenanceCronJob` to update correct and incorrect image and memory values
- [x] Added the function `ResetMaintenanceJobFrequency` to reset the cron expression to original one

**Which issue(s) this PR fixes** (optional)
Closes #PB-7090

**Special notes for your reviewer**:
[Jenkins BYOC job with sample test](https://jenkins.pwx.dev.purestorage.com/job/Users/job/Mithun/job/Quick%20BYOC/100/console)

### CODE REVIEW

File: tests/backup_helper.go [hunk 1]
```diff
@@ -8842,3 +8868,62 @@ func GetCronJobByBackupLocation(backupLocationName string, maintenanceJobType Ma
...
+type MaintenanceJobUpdateStrategy string
+
+const (
+	UpdateIncorrectImage  MaintenanceJobUpdateStrategy = "UpdateIncorrectImage"
+	UpdateCorrectImage    MaintenanceJobUpdateStrategy = "UpdateCorrectImage"
+	UpdateIncorrectMemory MaintenanceJobUpdateStrategy = "UpdateIncorrectMemory"
+	UpdateCorrectMemory   MaintenanceJobUpdateStrategy = "UpdateCorrectMemory"
+)
+
+func UpdateMaintenanceCronJob(backupLocation string, maintenanceJobType MaintenanceJobType, strategy MaintenanceJobUpdateStrategy) error {
```
Comment (kshithijiyer-px) [inline tests/backup_helper.go:8882]: Add docstring for the function

File: tests/backup/backup_sanity_test.go [hunk 2]
```diff
@@ -385,6 +385,47 @@ var _ = Describe("{ModifyCronJobFrequency}", func() {
 			dash.VerifyFatal(err, nil, "Modifying cron job frequency")
 			err = ModifyMaintenanceJobFrequency("mkoppal-nfs", 8, FullMaintenanceJob)
 			dash.VerifyFatal(err, nil, "Modifying cron job frequency")
+
+			err = ResetMaintenanceJobFrequency("mkoppal-nfs", QuickMaintenanceJob)
```
[the remark this answers is not in the document]
Comment (mkoppal-px) [reply tests/backup/backup_sanity_test.go:389]: I will remove this test case altogether. This is just a dummy test to try out the helper functions

File: tests/backup_helper.go (hunk 1 above)
Comment (mkoppal-px) [reply tests/backup_helper.go:8883]: Done

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**